Storage/SQL/SQLiteHistory.swift

```diff
@@ -451,46 +451,28 @@ extension SQLiteHistory: BrowserHistory {
             args = []
         }
 
-        let ungroupedSQL = [
-        "SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url, title, guid, domain_id, domain,",
-        "COALESCE(max(case \(TableVisits).is_local when 1 then \(TableVisits).date else 0 end), 0) AS localVisitDate,",
-        "COALESCE(max(case \(TableVisits).is_local when 0 then \(TableVisits).date else 0 end), 0) AS remoteVisitDate,",
-        "COALESCE(count(\(TableVisits).is_local), 0) AS visitCount",
-        "FROM \(TableHistory)",
-        "INNER JOIN \(TableDomains) ON \(TableDomains).id = \(TableHistory).domain_id",
-        "INNER JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id",
+        let sql = [
+        "SELECT",
+            "history.id AS historyID, history.url, title, guid, domain_id, domain,",
+            "COALESCE(MAX(CASE visits.is_local WHEN 1 THEN visits.date ELSE 0 END), 0) AS localVisitDate,",
+            "COALESCE(MAX(CASE visits.is_local WHEN 0 THEN visits.date ELSE 0 END), 0) AS remoteVisitDate,",
+            "COALESCE(COUNT(visits.is_local), 0) AS visitCount",
+            includeIcon ? ", iconID, iconURL, iconDate, iconType, iconWidth" : "",
+        "FROM",
+            "history",
+                "INNER JOIN domains ON domains.id = history.domain_id",
+                "INNER JOIN visits ON visits.siteID = history.id",
+                includeIcon ? "LEFT OUTER JOIN view_history_id_favicon ON view_history_id_favicon.id = history.id" : "",
         whereClause,
         "GROUP BY historyID",
-        ].joined(separator: " ")
-
-        let historySQL = [
-        "SELECT historyID, url, title, guid, domain_id, domain, visitCount,",
-        "max(localVisitDate) AS localVisitDate,",
-        "max(remoteVisitDate) AS remoteVisitDate",
-        "FROM (", ungroupedSQL, ")",
-        "WHERE (visitCount > 0)",    // Eliminate dead rows from coalescing.
-        "GROUP BY historyID",
-        "ORDER BY max(localVisitDate, remoteVisitDate) DESC",
+        "HAVING COUNT(visits.is_local) > 0",
```

Comment:
> The original query was doing this too…

Then I'm confused, because I see remote-only history in my history panel, and yet this query would seem to exclude them!